Bio/SeqUtils/MeltingTemp.py

```diff
@@ -442,7 +444,8 @@ def _check(seq, method):
     'ACGTTGCAAGTCCATGGTAC'
 
     """
-    seq = "".join(seq.split()).upper()
+    # Remove all non-alphabetic characters
+    seq = re.sub(r"[^A-Z]", "", seq.upper())
```

Comment:
This change is most contentious - and ought to be in the docstring if we do go ahead with it. Some punctuation is perhaps reasonable in a sequence (I am thinking of gaps in particular), but personally in the context of thermodynamic calculations I think they ought to raise an exception.

Reply:
@peterjc I agree with raising exceptions, but that's how it was working before, they were just being removed in later code by the `_check` function.

Reply:
Not sure if you think someone was using it like that. Happy to make it more strict.

Reply:
I would be against silently removing all non-alphabetic characters, as it can mask bugs.

Removing non-alphabetic characters from the sequence should be left to the user (before calling the functions in `MeltingTemp`), as only the user can know which non-alphabetic characters (if any) may appear in the sequence.

Putting a warning in the docstring is not sufficient. People are likely to miss it, or to not read the docstring at all. If the non-alphabetic characters appear in the sequence due to some bug in the user's script, the user will not be aware of it and may think that the warning in the docstring is not relevant.

Reply:
The use case here are simple strings. The main purpose of this `_check()` function is to remove typical artifacts of copy-pasted or simple file read sequences that may contain whitespace, digits or gap characters, e.g. `1 ATGCATGCAT GC 12` or `ATGC--ATGC`. I want to allow such kind of input.

Reply:
Like Michiel, I favour the strict approach. Given the bug report this was too lax historically. Would a warning on non-letter content be a compromise?

Reply:
This is even more lax than the original code.

Reply:
You have to look at it in the context of (original) line 471. Here's the original two lines in question:
```
seq = "".join(seq.split()).upper()
...
seq = "".join([base for base in seq if base in baseset])
```
That second line is doing the same thing as this sub, except it's looser because it just deletes alphabetic characters outside of baseset instead of throwing an error. In the PR version we have:
```
seq = re.sub(r"[^A-Z]", "", seq.upper())
...
if not all(base in baseset for base in seq):
    raise ValueError("The input sequence is not valid for " + method)
```
so, unlike the original, it will at least throw an error if there are alphabetic characters outside the base case